lib/Pool.js

```diff
@@ -259,13 +261,21 @@ class Pool {
   }
   _destroy(resource) {
     try {
-      this.destroyer(resource);
+      const retVal = this.destroyer(resource);
+      if (retVal && retVal.catch) {
+        // There's nothing we can do here but log the error.
+        retVal.catch(this._logError.bind(this));
+      }
+      return retVal;
```

Comment:
This part I don't understand. Assuming `destroyer` is a promise now returning the value like this makes the try-catch pretty redundant since it's not 'awaited' for.

Reply:
Isn't this equivalent to the current catch behaviour? While we don't await for it, I guess we still want to log it to not hide it from the user (or at least that's why I assume we're logging the error for the sync `destroyer`)

Reply:
What I mean is calling try/catch on a promise without async/await is redudant. Error is caught in .catch handler instead. Can remove try catch block completely.

Reply:
it's meant to catch synchronous errors caused by `this.destroyer` - since that's the old behaviour. Should I make a comment into the catch block to make that clearer?

Reply:
Since destroy is returning promise now shouldn't it convert that catched error to caller as `return Promise.reject(err)`? Also I'm not sure at all if _destroy() should catch all error internally anymore...

Reply:
I'm happy to make that change - I have no insight into why the error was silenced in the sync case. could you as the maintainers decide what you'd like here?

Reply:
For now, I will keep the current behaviour - i.e., hide errors & always resolve
Let me know if you want to change that